### ISSUE TITLE
Incorrect "Missing semicolon" warning on a one-line function

### ISSUE THREAD

**Issue by mamacdon** [eslint/eslint#121]
Using ESLint 0.0.8-dev, when I run the `semi` rule on this code:

``` javascript
setTimeout(function() {foo = "bar";});
```

I see a warning on the function expression.
`line 1, col 23, Warning - Missing semicolon.`

This is wrong, since the code has all the semicolons it needs. FWIW, if you insert a newline before the function's closing '}', the warning disappears.
